### DISCUSSION TITLE
From J.2000 to On Date Coordinates of a Star.

### DISCUSSION THREAD

**Discussion by BikramAjay** [Stellarium/stellarium#1819] [General]
What are the transformations applied on the RA/Dec(J.2000) of a star to translate it to "On date"? 
Is it just Precession + Nutation or something more than it? Can't figure out from Stelcore.cpp and other related files hence asking. 
I wrote a piece of code implementing this transformation based on precession and nutation matrices only. The problem is that it's output starts differing from Stellarium's "On Date" values when I go back in time more than 1000 years. I was wondering what I am missing. 

![Screenshot 2021-07-27 095311](https://user-images.githubusercontent.com/86772525/127166893-ab8b70b7-6e83-4bae-bf90-f813b41fde5d.jpg)

**Comment by gzotti** [comment]
Proper motion, precession+nutation, refraction. 

Models are given in the User Guide, Appendix F. 

As written in the User Guide, Appendix F.5, we compute nutation only +/-500 years around J2000 because the valid timespan of IAU 2000B model is unknown.

I am currently working on finally adding annual aberration, but have too little time this week to continue.

**Reply by BikramAjay** [reply]
Thanks for the clarification. I was wondering how do you combine proper motion and precession. Do you apply proper motion correction on the rotated (due to precession) coordinates or do you apply proper motion correction first and then rotate it for precession correction?

**Reply by gzotti** [reply]
Please see the ZoneArray code, it predates my affiliation. They are called "PM components in J2000 coordinates".  So, we first compute PM for epoch E in the J2000 frame, then apply whatever else is necessary to J2000 coordinates, like applying precession/nutation from J2000 to E. But this is algorithmic info better found in textbooks. Compare your results also to certified reference data!

**Reply by BikramAjay** [reply]
Thanks.
